### ISSUE TITLE
MODIFY_INCREMENT & controls with mock sync strategy doesn't work

### ISSUE THREAD

**Issue by fsaintma** [cannatag/ldap3#887]
Hi

I use mock sync strategy (i.e ldap3==2.8.1) and my test calls modify operation with MODIFY_INCREMENT changes. It seems that the operation is not done even if we obtained a succes result. Moreover when we add controls (POST_READ_CONTROL) the controls dictionnary is not  returned . I provide you the following code which shows the bug. Of course with a connection to a real OpenLDAP server we have the expected behaviour.

Thanks for your help.

```
#! /usr/bin/env python
# -*- coding:utf-8 -*

from ldap3 import Connection, MOCK_SYNC, Server
from ldap3 import MODIFY_REPLACE, MODIFY_INCREMENT
from ldap3.protocol.rfc4527 import post_read_control

fake_server = Server('my_fake_server')
fake_conn = Connection(fake_server,
                       'uid=test,dc=test,dc=fr',
                       'test',
                       client_strategy=MOCK_SYNC)
fake_conn.strategy.add_entry('uid=test,dc=test,dc=fr',
                             {'userPassword': 'test'})

fake_conn.strategy.add_entry('cn=uidNext,dc=test,dc=fr',
                             {'uidNumber': 5000, 'objectclass': ['top']})

with fake_conn as conn:
     conn.modify('cn=uidNext,dc=test,dc=fr',
                 changes={'uidNumber': [(MODIFY_REPLACE, [5001])]})
     print(conn.result)
     conn.search('cn=uidNext,dc=test,dc=fr', '(objectClass=*)',
                 attributes=['uidNumber'])
     print(conn.entries[0].uidNumber.value)

with fake_conn as conn:
     conn.modify('cn=uidNext,dc=test,dc=fr',
                 changes={'uidNumber': [(MODIFY_INCREMENT, [1])]})
     print(conn.result)
     conn.search('cn=uidNext,dc=test,dc=fr', '(objectClass=*)',
                 attributes=['uidNumber'])
     print(conn.entries[0].uidNumber.value)

with fake_conn as conn:
     conn.modify('cn=uidNext,dc=test,dc=fr',
                 changes={'uidNumber': [(MODIFY_INCREMENT, [1])]},
                 controls=[post_read_control('gidNumber')])
     print(conn.result)
     conn.search('cn=uidNext,dc=test,dc=fr', '(objectClass=*)',
                 attributes=['uidNumber'])
     print(conn.entries[0].uidNumber.value)
     # expect
     # conn.result['controls']['1.3.6.1.1.13.2']['value']
     #            ['result']['gidNumber'][0])
```
The output:

```
{'result': 0, 'description': 'success', 'message': '', 'dn': '', 'referrals': None, 'type': 'modifyResponse'}
5001
{'result': 0, 'description': 'success', 'message': '', 'dn': '', 'referrals': None, 'type': 'modifyResponse'}
5001
{'result': 0, 'description': 'success', 'message': '', 'dn': '', 'referrals': None, 'type': 'modifyResponse'}
5001
```

**Comment by cannatag** [comment]
Hi, sorry for answering late, can you check the code in the dev branch? It should work now.

Thanks,
Giovanni

**Comment by cannatag** [comment]
Fixed in ldap3 v2.9
